ui/tests/integration/components/kv/page/kv-page-version-history-test.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { setupEngine } from 'ember-engines/test-support';
+import { setupMirage } from 'ember-cli-mirage/test-support';
+import { click, render } from '@ember/test-helpers';
+import { hbs } from 'ember-cli-htmlbars';
+import { kvMetadataPath } from 'vault/utils/kv-path';
+import { PAGE } from 'vault/tests/helpers/kv/kv-page-selectors';
+import { allowAllCapabilitiesStub } from 'vault/tests/helpers/stubs';
+
+module('Integration | Component | kv | Page::Secret::Version-History', function (hooks) {
+  setupRenderingTest(hooks);
+  setupEngine(hooks, 'kv');
+  setupMirage(hooks);
+
+  hooks.beforeEach(async function () {
+    const store = this.owner.lookup('service:store');
+    this.server.post('/sys/capabilities-self', allowAllCapabilitiesStub());
+    const metadata = this.server.create('kv-metadatum');
+    metadata.id = kvMetadataPath('kv-engine', 'my-secret');
```

Comment:
I think the metadata should be changed here, to test that both icons render here, instead of changing the factory (which will just mean we keep having to update tests that use it)

```js
 metadata.versions["4"] = {
      created_time: '2023-07-21T03:11:58.095971Z',
      deletion_time: '',
      destroyed: true,
}